File: vimeo-networking/src/main/java/com/vimeo/networking/model/Entity.java
package com.vimeo.networking.model;

import org.jetbrains.annotations.Nullable;

/**
 * Represents data that can be uniquely identified by an {@link Entity#getIdentifier()}.
 * <p>
 * Implementors of this interface must provide a unique identifier.
 */
public interface Entity {

    /**
     * The identifier, as a {@link String}, that uniquely identifies this entity.
     *
     * @return a {@link String}, or null if the entity does not have an identifier.
     */
    @Nullable
    String getIdentifier();

}

File: models/src/main/java/com/vimeo/networking2/common/Entity.kt
package com.vimeo.networking2.common

/**
 * Represents data that can be uniquely identified by an [Entity.identifier].
 *
 * Implementors of this interface must provide a unique identifier.
 */
interface Entity {

    /**
     * The identifier, as a [String], that uniquely identifies this entity.
     *
     * @return a [String], or null if the entity does not have an identifier.
     */
    val identifier: String?

}
